.github/workflows/pr_ci.yml

```diff
@@ -15,17 +15,15 @@ jobs:
     name: Build & Test
     steps:
     - uses: actions/checkout@v2
-    - uses: actions/setup-node@v2
+    - uses: actions/cache@v3
       with:
-        node-version: "15"
-    - name: Setup .NET 6.0
-      uses: actions/setup-dotnet@v1
-      with:
```

Comment:
I'm not sure about these. While it's great to save on having to do these, but we also take a dependency on `ubuntu-latest` and `windows-latest` containing the version we want, and if the images are updated, our build can break without us touching it. Perhaps these actions could instead be optimized not to do the installation if it's already available?

Reply:
Could you reply, please?

Reply:
Definitely no need to install Node 15 when 16 is LTS, seems that for Windows image every unnecessary step will be slow. Using `-latest` already makes building unpredictable - it can be Windows 2032 some day.

Reply:
We definitely shouldn't use v15 of Node, I agree. However, that doesn't necessarily mean we should use the version provided by the image, which is not really something contributors would keep in mind. While using the "latest" images indeed poses a similar issue, I'd say that the chance that a Node or .NET SDK version update brings issues is at least an order of magnitude larger, than that of other image updates, especially OS ones. It's anecdotal, but we even had minor .NET SDK versions breaking the build here (and I've seen that many times with our other apps at Lombiq too), while I can't remember OS updates breaking anything like this.

Similar to `setup-dotnet`, `setup-node`
[won't install anything if the version already exists locally](https://github.com/actions/setup-node#:~:text=The%20action%20will%20first%20check%20the%20local%20cache%20for%20a%20semver%20match.%20If%20unable%20to%20find%20a%20specific%20version%20in%20the%20cache%2C%20the%20action%20will%20attempt%20to%20download%20a%20version%20of%20Node.js.). So if we target v16 (since [the 3 latest LTS versions are pre-installed](https://github.com/actions/runner-images#:~:text=3%20latest%20LTS%20versions)) then this shouldn't be significantly slower than not doing anything, while retaining that the version we use for the build is known.